### PR TITLE
feat/dev-tools: PHPStan, PHPInsights & Laravel Pint

### PR DESCRIPTION
**Pull request by joelbutcher**
## TL;DR
Adds dev tooling to cachet/core

## Description
This PR adds the following dev tools into the cachet/core project

- PHPStan (via Larastan), set to Level 5 – this can be incremented gradually if we want a higher level of strictness
- PHPInsights – configured to the "laravel" preset with "phpstorm" set as the IDE
- Laravel Pint – default preset, could be changed / adjusted as required

### Important things to note
- I've updated all PHP files to use `declare(strict_types=1);`
- I've added `format` (pint), `analyse` (phpstan) and `complexity` scripts to the `composer.json`
- I've **not** added any checks into CI (yet)

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Thanks for this, @joelbutcher. However, I'm not going to merge this at this time. I really want to nail the features and design implementation before we get into any of this.

Note that Pint runs automatically when anything is merged into `main`.

Let's keep PHPStan in mind for the future though.
